integration-tests/utils/firecloud-utils.js

```diff
@@ -7,28 +7,39 @@ const selectWorkspace = async (page, billingAccount, workspace) => {
 }
 
 const signIntoFirecloud = async (page, token) => {
+  await page.waitForXPath('//title[text()="FireCloud | Broad Institute"]')
+  await findText(page, 'content you are looking for is currently only accessible')
+
+  await page.waitForXPath('//*[@id="sign-in-button"]')
+
   /*
    * The FireCloud not-signed-in page renders the sign-in button while it is still doing some
    * initialization. If you log the status of the App components state for user-status and auth2
    * with each render, you see the following sequence:
    *   '#{}' ''
    *   '#{}' '[object Object]'
    *   '#{:refresh-token-saved}' '[object Object]'
+   *
    * If the page is used before this is complete (for example window.forceSignedIn adding
    * :signed-in to user-status), bad things happen (for example :signed-in being dropped from
-   * user-status). Instead of reworking the sign-in logic for a case that (for the most part) only
-   * a computer will operate fast enough to encounter, we'll just slow the computer down a little.
+   * user-status). Put another way, if forceSignedIn is called between the 2nd and 3rd render, the
+   * sign-in doesn't work. The visible effect (failure screenshot) is that the test gets stuck on
+   * the FireCloud sign-in page.
+   *
+   * There is no observable evidence that the component has settled into its final state. I (Brian)
+   * believe that watching for the sign-in button (which is also what the FireCloud UI tests did:
+   * https://github.com/broadinstitute/firecloud-ui/blob/ab678610367c8388d70de768db0ad84c6df63997/automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala#L31) will get us past the 2nd render
+   * but not necessarily the thrid. Therefore, instead of reworking the sign-in logic for a case
+   * that (for the most part) only a computer will operate fast enough to encounter, we'll just slow
+   * the computer down a little with a fixed-length sleep.
+   *
+   * Note that the amount of sleep needed is dependent on CPU load where the browser is running,
+   * which can be affected by CPU speed, memory, and other processes running at the same time
+   * (notably tests running in parallel). The duration below may be overkill in many conditions, but
+   * we want it to always be enough, whether running 10 parallel executions on a developer laptop or
+   * running in a CI server.
    */
-  await page.waitForResponse(response => {
```

Comment:
@aweng98, I really did like this approach, but as I described in the PR description, the difference between headless and non-headless was a deal-breaker for me. Also, neither this nor searching for the sign-in button are alone sufficient to guarantee that the page is ready. Since we can't completely eliminate the `delay`, I chose the sign-in button approach because I found it to be more reliable.

I'm looking forward to hearing your thoughts about this PR. Thanks.

Reply:
Wait for the `sign-in-button` is a good call.
There was a 1 sec sleep before, but I removed it in this [PR](https://github.com/DataBiosphere/terra-ui/pull/3036) because it didn't eliminate flaky failure. 
Maybe, waiting for `sign-in-button` and `delay(1500)` is going to remove the flakiness for good.
Since you have determined this is not a product or UI code issue, I think adding a [retry](https://github.com/DataBiosphere/terra-ui/commit/cab27f927dfa8a6dd58f4afb71a3466c104a01d5) will guarantee no more flaky failure. Or maybe we add it in future if flakiness continues.